Bucardo.pm

```diff
@@ -6934,38 +6934,37 @@ sub validate_sync {
 
 	# Table cache
     $SQL{checktableonce} = q{
-            SELECT n.nspname, c.relname, c.oid,quote_ident(n.nspname) as safeschema, quote_ident(c.relname) as safetable, quote_literal(n.nspname) as safeschemaliteral, quote_literal(c.relname) as safetableliteral
+            SELECT n.nspname, c.relname, c.oid, quote_ident(n.nspname) as safeschema, quote_ident(c.relname) as safetable, quote_literal(n.nspname) as safeschemaliteral, quote_literal(c.relname) as safetableliteral
             FROM   pg_class c, pg_namespace n
             WHERE  c.relnamespace = n.oid
         };
     $sth = $srcdbh->prepare($SQL{checktableonce});
 	$sth->execute();
     my %tablescache;
 	for my $row (@{$sth->fetchall_arrayref({})}) {
-		$tablescache{$row->{nspname}.$row->{relname}}{oid}=$row->{oid};
-		$tablescache{$row->{nspname}.$row->{relname}}{safeschema}=$row->{safeschema};
-		$tablescache{$row->{nspname}.$row->{relname}}{safetable}=$row->{safetable};
-		$tablescache{$row->{nspname}.$row->{relname}}{safeschemaliteral}=$row->{safeschemaliteral};
-		$tablescache{$row->{nspname}.$row->{relname}}{safetableliteral}=$row->{safetableliteral};
+        $tablescache{"$row->{nspname}.$row->{relname}"} = {
+            map { $_ => $row->{$_} } qw(oid safeschema safetable safeschemaliteral safetableliteral)
+        };
     }
 	$sth->finish();
 
     GOAT: for my $g (@{$s->{goatlist}}) {
 
         ## TODO: refactor with work in validate_sync()
 
-        $self->glog(qq{  Inspecting source $g->{reltype} "$g->{schemaname}.$g->{tablename}" on database "$sourcename"}, LOG_NORMAL);
+        my $t = "$g->{schemaname}.$g->{tablename}";
+        $self->glog(qq{  Inspecting source $g->{reltype} "$t" on database "$sourcename"}, LOG_NORMAL);
         ## Check the source table, save escaped versions of the names
 
-        if (!exists ($tablescache{$g->{schemaname}.$g->{tablename}})) {
-            my $msg = qq{Could not find $g->{reltype} "$g->{schemaname}"."$g->{tablename}"\n};
+        if (!exists ($tablescache{$t})) {
+            my $msg = qq{Could not find $g->{reltype} "$t"\n};
```

Comment:
The one issue I have with this patch (minor) is the `$msg` var is quoting the entirety of the schema/table, rather than quoting the schema and the table separately, which in the message might end up being a little misleading.  However this does fix some other bugs, so happy to overlook it for now.  (I would expect in practice if it needs to quote either part of the schema/table that the message would look something like: `Could not find table ""quoted schema"."Table""` or similar, so not worth trying to kick back for exact correctness for that reason.)